db/migrate/20200519213913_change_hardware_mem_types.rb

```diff
@@ -0,0 +1,17 @@
+class ChangeHardwareMemTypes < ActiveRecord::Migration[5.2]
+  class Hardware < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  def up
+    change_column :hardwares, :memory_mb, :bigint
+    rename_column :hardwares, :memory_mb, :ram_size_in_bytes
+    Hardware.update_all("ram_size_in_bytes = ram_size_in_bytes * 1048576")
```

Comment:
```suggestion
    Hardware.update_all("ram_size_in_bytes = COALESCE(ram_size_in_bytes * 1048576, 0)")
```

we'll get rid of the problematic null values.
and adding the default_value_for will fix it for already created records.